### ISSUE TITLE
Android: ogg build error

### ISSUE THREAD

**Issue by nerzhul** [minetest/minetest#7409]
##### Issue type

 - Build issue
 
##### Minetest version

```
0.4.17
```

##### OS / Hardware

Operating system: Archlinux
CPU: AMD FX-8350

GPU model: Radeon Vega RX580
OpenGL version: 4.5

##### Summary

Android build doesn't work when trying to build libogg

```
[armeabi-v7a] SharedLibrary  : libvorbis-jni.so
xxx/android-sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld: warning: skipping incompatible /usr/lib/libatomic.so while searching for atomic
xxx/android-sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld: error: treating warnings as errors
collect2: error: ld returned 1 exit status
```

This blocks the 0.4.17 release for the play store

**Comment by nerzhul** [comment]
i will try to backport the Makefile from master, which uses clang, it should run fine and work with atom

**Comment by stujones11** [comment]
Strangely b5350e27ad7bc54a3a97db988119db83c67f96b3 builds and runs just fine on my device except for this warning: `libnativeloader.so unauthorized access to /system/lib/libssl.so` IIRC there was an easy fix for this.

**Comment by stujones11** [comment]
I believe this fix for this warning was to remove the following lines
https://github.com/minetest/minetest/blob/b5350e27ad7bc54a3a97db988119db83c67f96b3/build/android/src/main/java/net.minetest.minetest/MtNativeActivity.java#L86-L87

**Comment by MoNTE48** [comment]
https://github.com/MoNTE48/libvorbis-android
https://github.com/MultiCraftProject/MultiCraft/blob/master/build/android/Makefile#L225
https://github.com/MultiCraftProject/MultiCraft/blob/master/build/android/jni/Android.mk#L32
_I'll just leave it here
Nobody needs this._

**Comment by MoNTE48** [comment]
Android port does not require ssl. Let the one who does not agree, explain why he needs an Android player! @stujones11 +1!
I (or @sfan5) can transfer multiple patches from MultiCraft (or make them myself) to Minetest and release a really good 0.4.17 for Android!
Or just fix the build error and not pay attention to me for a few more years. Anyway, in the original minetest on Android, no one will play.

**Comment by nerzhul** [comment]
@stujones11 the flag doesn't work well

**Comment by nerzhul** [comment]
issue is fixed in stable-0.4 by using the proper toolchain
